### PR TITLE
Feature/add plugable status labels

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
This classes can be used to add any sort of panel to the right side of DC as info Panels. See DataCleanerCloud panel or News Items

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 1]
```diff
@@ -255,6 +260,8 @@ public void onRemove(final InputColumn<?> sourceColumn) {
 
     private static final Logger logger = LoggerFactory.getLogger(AnalysisJobBuilderWindow.class);
     private static final ImageManager imageManager = ImageManager.get();
+    
+    public static final List<Func<AnalysisJobBuilderWindowImpl, JComponent>> PLUGGABLE_LABELS_BANNER_COMPONENTS = new ArrayList<>(0);
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: I know this is how we've done it in other places, but I'm really no fan of exposing what should be internal structures like this. Can't we use the ServiceLoader interface, or some other proper solution instead?

At least hide it behind an `addRightPanelComponent()` or something like that.

The name also seems a bit weird. It's not a banner?

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 2]
```diff
@@ -765,6 +772,17 @@ private JComponent getWindowPanelContent() {
         final JXStatusBar statusBar = WidgetFactory.createStatusBar(_statusLabel);
         RightInformationPanel rightInformationPanel = new RightInformationPanel(_glassPane);
 
+        for (Func<AnalysisJobBuilderWindowImpl, JComponent> pluggableComponent : PLUGGABLE_LABELS_BANNER_COMPONENTS) {
+            final JComponent component = pluggableComponent.eval(this);
+            if (component != null) {
+                if (component instanceof JPanel) {
+                    final PlugabblePanel panel = (PlugabblePanel) component;
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: Checking for an instance of a JPanel, but casting to a PluggablePanel seems wrong to me.

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 3]
```diff
@@ -765,6 +772,17 @@ private JComponent getWindowPanelContent() {
         final JXStatusBar statusBar = WidgetFactory.createStatusBar(_statusLabel);
         RightInformationPanel rightInformationPanel = new RightInformationPanel(_glassPane);
 
+        for (Func<AnalysisJobBuilderWindowImpl, JComponent> pluggableComponent : PLUGGABLE_LABELS_BANNER_COMPONENTS) {
+            final JComponent component = pluggableComponent.eval(this);
+            if (component != null) {
+                if (component instanceof JPanel) {
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: `null` wouldn't be an instance of JPanel, so the null check is a bit redundant

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java (hunk 1 above)
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: Oh, and you should never pass the Impl classes. Use the interface, that's what it is there for.

Comment (ClaudiaPHI) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: I could use the ServiceLoader, but wouldn't that be a mix of a class in the enterprise edition with the community edition ?

Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: Not sure I understand? Using the ServiceLoader, you would provide an interface that any JAR could then implement. Then DataCleaner would just load all available implementations, and add them by itself.

For inspiration, see how we provide an implementation of SparkJobLifeCycleListener in the commercial spark package, and add it in `org.datacleaner.spark.Main` line 67.

Of course, this being a Swing class, the actual service needs to be some kind of factory that provides the `PluggablePanel` in `getPluggablePanel()` or similar method.

**Comment by ClaudiaPHI** [comment]
Ready for review again
2016-09-30 15_32_35-welcome _ datacleaner
2016-09-30 15_22_03-welcome _ datacleaner

File: desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java [hunk 4]
```diff
@@ -773,9 +776,20 @@ private JComponent getWindowPanelContent() {
...
-
-        final LicenceAndEditionStatusLabel statusLabel = new LicenceAndEditionStatusLabel(rightInformationPanel);
-        statusBar.add(statusLabel);
+        
+        if (Version.isCommunityEdition()) {
+            final CommunityEditionStatusLabel statusLabel = new CommunityEditionStatusLabel(rightInformationPanel);
+            statusBar.add(statusLabel);
+            statusBar.add(Box.createHorizontalStrut(20));
+        } else {
+            final ServiceLoader<PlugabblePanel> pluggablePanelsLoaders = ServiceLoader.load(PlugabblePanel.class);
```
Comment (LosD) [inline desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java]: Hmmmm... Something seems a bit off here. You should probably make a factory/descriptor for the icon and panel in the factory, like:
```
interface InformationPanelDescriptor {
    JComponent createInformationComponent(AnalysisJobBuilderWindow analysisJobBuilderWindow);
    String getLabelImagePath();
    String getLabelText();
}
```

... And load that using the service loader instead. That also removes the dependency on DCPanel, any JComponent is (and should be) usable with the RightInformationPanel.

I'd maybe also drop the PluggableRightPanelLabel name... The label itself is not pluggable, and the position can change (yes, I know RightInformationPanel is already misnamed in that regard, but let's not make it worse :)). How about just InformationPanelLabel?

**Comment by LosD** [comment]
LGTM!

**Pull request merged**